parsley-debug/shared/src/main/scala/parsley/debug/util/Collector.scala

```diff
@@ -35,8 +35,18 @@ object Collector {
         Renamer.addNames(XCollector.collectNames(obj))
     }*/
 
-    /** This is an internal method used by the `parsley.debuggable` annotation */
-    def registerNames(names: Map[Parsley[_], String]): Unit = {
+    /**
+      * This is an internal method used by the `parsley.debuggable` annotation. 
+      * 
+      * This function registers the names collected by the annotation to the 
+      * `Renamer` object, and also adds the source file path the annotation was
+      * found inside of the the `SourceCollector` object.
+      *
+      * @param names List of parsers mapped to their string representations.
+      * @param source Source file the parsers are being added from.
+      */
+    def registerNames(names: Map[Parsley[_], String], source: String): Unit = {
```

Comment:
If you're not planning on supporting this on Scala 2, you should make this an `Option[String]`